### PR TITLE
Add accessibility render extension to render metadata for accessibility

### PR DESCRIPTION
**Pull request by geoff-powell**
Based on https://github.com/cashapp/paparazzi/pull/184

This pr builds on top of the render extension api to add simple rendering of accessibility information to view snapshots. This was initially inspired by [Accessibility Snapshot](https://github.com/cashapp/AccessibilitySnapshot) for ios.

## Example
- Sample Test - [link](https://github.com/cashapp/paparazzi/blob/380de34027f85f06146cbf92f3e74db2c5785532/paparazzi/extensions/accessibility/src/test/java/app/cash/paparazzi/extensions/accessibility/AccessibilityRenderExtensionTest.kt)
6ba3fb61f7b66371790b94604b4fe3ea70ab298e

### CODE REVIEW

File: paparazzi/extensions/accessibility/build.gradle [hunk 1]
```diff
@@ -0,0 +1,41 @@
+apply plugin: 'java-library'
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:1]: not needed?

File: paparazzi/extensions/accessibility/build.gradle [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+apply plugin: 'kotlin'
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:2]:
```suggestion
apply plugin: 'org.jetbrains.kotlin.jvm'
```

File: paparazzi/extensions/accessibility/build.gradle [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+apply plugin: 'org.jetbrains.kotlin.kapt'
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:3]: prob need to add `apply plugin: 'com.vanniktech.maven.publish'` now?

File: paparazzi/extensions/accessibility/build.gradle [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:6]: not needed?

File: paparazzi/extensions/accessibility/build.gradle [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      writer.writeLine(".")
+      writer.writeLine("29")
+      def androidHome = System.getenv("ANDROID_SDK_ROOT")
+          ?: System.getenv("ANDROID_HOME")
+          ?: "$userHome/Library/Android/sdk"
+      writer.writeLine("${androidHome}/platforms/android-29/")
+      writer.writeLine(".")
+    }
+  }
+}
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:33]: not needed?

File: paparazzi/extensions/accessibility/build.gradle [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+tasks.withType(Test).configureEach { task ->
+  task.dependsOn(generateTestConfig)
+  task.systemProperty(
+      "paparazzi.test.resources",
+      generateTestConfig.map { it.outputs.files.singleFile }.get().path
+  )
+}
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:41]: not needed?

File: paparazzi/extensions/accessibility/src/main/java/app/cash/paparazzi/extensions/accessibility/RenderSettings.kt [hunk 7]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/src/main/java/app/cash/paparazzi/extensions/accessibility/RenderSettings.kt:2]: use current year in all new copyrights

File: paparazzi/extensions/accessibility/src/test/java/app/cash/paparazzi/R.kt [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package app.cash.paparazzi
+
+/** Simulate an empty R.java for this package. */
+class R
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/src/test/java/app/cash/paparazzi/R.kt:4]: why is this needed?

File: paparazzi/extensions/accessibility/build.gradle [hunk 9]
```diff
@@ -0,0 +1,41 @@
+apply plugin: 'java-library'
+apply plugin: 'kotlin'
+apply plugin: 'org.jetbrains.kotlin.kapt'
+
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
+
+dependencies {
+  api project(':paparazzi')
```
Comment (jrodbx) [inline paparazzi/extensions/accessibility/build.gradle:9]: what types does this need from paparazzi?  might be worth pulling into a commons?

File: settings.gradle [hunk 10]
```diff
@@ -2,6 +2,7 @@ rootProject.name = 'paparazzi-parent'
 
 include ':sample'
 include ':paparazzi'
+include ':paparazzi:extensions:accessibility'
```
Comment (jrodbx) [inline settings.gradle:5]: paparazzi-accessibility?

File: paparazzi/src/main/java/app/cash/paparazzi/RenderExtension.kt [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+/**
+RenderExtension is an interface that can render additional information on top of the ui Paparazzi renders for each frame
+ */
+interface RenderExtension {
+  fun render(
+    snapshot: Snapshot,
+    view: View,
+    image: BufferedImage,
+  ): BufferedImage
+}
```
Comment (jrodbx) [inline paparazzi/src/main/java/app/cash/paparazzi/RenderExtension.kt:30]: ah, is this the needed type?  maybe this is what goes into its own module then?

**Review by jrodbx — CHANGES_REQUESTED** [review]

File: paparazzi/extensions/accessibility/src/test/java/app/cash/paparazzi/R.kt (hunk 8 above)
Comment (geoff-powell) [reply paparazzi/extensions/accessibility/src/test/java/app/cash/paparazzi/R.kt:4]: Its needed for the unit test

File: .github/workflows/build.yml [hunk 12]
```diff
@@ -62,7 +62,9 @@ jobs:
           java-version: 11
 
       - name: Publish Artifacts
-        run: ./gradlew publishMavenPublicationToMavenCentralRepository paparazzi-gradle-plugin:publish
+        run: |
+          ./gradlew publishMavenPublicationToMavenCentralRepository paparazzi-gradle-plugin:publish
+          ./gradlew publishMavenPublicationToMavenCentralRepository paparazzi-accessibility:publish
```
Comment (geoff-powell) [inline .github/workflows/build.yml:67]: @jrodbx Not as familure with the maven publish plugin we are using but I added this to make sure the artifact is also published

File: paparazzi-accessibility/build.gradle [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+      writer.writeLine(".")
+      writer.writeLine("29")
+      writer.writeLine("platforms/android-29/")
+      writer.writeLine(".")
+      writer.writeLine(configurations.unzip.singleFile.path)
+    }
+  }
+}
+
+// TODO: Remove this duplicate of logic in paparazzi/build.gradle
```
Comment (jrodbx) [inline paparazzi-accessibility/build.gradle:55]: Is depending on paparazzi module making this script boilerplate necessary?  If so, then we might have to go with the other modular structure after all?

Comment (geoff-powell) [reply paparazzi-accessibility/build.gradle:55]: Nope this is due to the unit tests I have in the module. I could move them to one of the ones run in the `paparazzi-gradle-plugin` and then we don't need any of the logic since it runs against paparazzi directly via the plugin`

Comment (geoff-powell) [reply paparazzi-accessibility/build.gradle:55]: So after testing this out it seems I'm going to need to invert the module dependency. I tried using the accessibility repo in the sample module and it was causing issues with the gradle plugin dependencies

**Review by jrodbx — APPROVED** [review]
nice!

**Comment by geoff-powell** [comment]
The diff is pretty small. Upping the test to 0.2% max.
delta-accessibility
